rflx/templates/rflx_generic_types.ads

```diff
@@ -26,6 +26,23 @@ is
 
    subtype Bit_Length is Custom_Bit_Length;
 
+   function "+" (Left : Index; Right : Length) return Index is
+      (Index (Length (Left) + Right)) with
+      Pre => Length (Left) <= Length'Last - Right;
+
+   function "+" (Left : Index; Right : Index) return Index is abstract;
+
+   function "-" (Left : Index; Right : Index) return Length is
+      (Length (Left) - Length (Right)) with
+      Pre => Length (Left) >= Length'First + Length (Right);
+
+   function "-" (Left : Index; Right : Index) return Index is abstract;
+
+   function "-" (Left : Index; Right : Length) return Index is
+      (Index (Length (Left) - Right)) with
+      Pre => Right < Length'Last
+             and then Length (Left) >= Length (Index'First) + Right;
```

Comment:
For consistency, it would be nice if you could put the `with` on a separate line.

Reply:
Fixed.

Reply:
I guess my comment was not very clear. Sorry about that. The `with` is now on a new line, but it is still not on a separate line. It would be nice if the style of this code would be consistent to the existing code. For example:

```ada
   function Extract
      (Buffer : Bytes_Ptr;
       First  : Index;
       Last   : Index;
       Off    : Offset;
       Size   : Positive;
       BO     : Byte_Order) return U64
   with
     Pre =>
       (Buffer /= null
        and then First >= Buffer'First
        and then Last <= Buffer'Last
        and then Size in 1 .. U64'Size
        and then First <= Last
        and then Last - First <= Index'Last - 1
        and then Length ((Offset'Pos (Off) + Size - 1) / Byte'Size) < Length (Last - First + 1)
        and then (Offset'Pos (Off) + Size - 1) / Byte'Size <= Natural'Size
        and then (Byte'Size - Natural (Offset'Pos (Off) mod Byte'Size)) < Long_Integer'Size - 1);
```

Could you please change the code again? Too bad that there is no good code formatter...

Reply:
I see. I took the style of some examples I found in the generated code. But taking a second look it seems like there is no consistency in the first place. From `generated/rflx-rflx_arithmetic.ads`:
```Ada
   --  Wrapper of Shift_Left that expresses the operation in terms of
   --  Fits_Into/Fits_Into_Upper.
   function Left_Shift (V : U64; Amount : Natural; Size : Natural) return U64 with
     Pre =>
       Size < U64'Size
       and then Amount < U64'Size
       and then Fits_Into (V, Size)
       and then Size + Amount < U64'Size,
       Post => Fits_Into_Upper (Left_Shift'Result, Size + Amount, Amount);

   --  V is assumed to have Bits bits of data. Set the lower bits of V to zero.
   function Mask_Lower (V : U64; Mask, Bits : Natural) return U64
     with Pre => Bits <= U64'Size and then Fits_Into (V, Bits) and then Mask <= Bits and then Mask >= 1,
          Post => Fits_Into_Upper (Mask_Lower'Result, Bits, Mask);

   --  Set the upper bits of V to zero.
   function Mask_Upper (V : U64; Mask : Natural) return U64
     with Pre => Mask < U64'Size,
          Post => Fits_Into (Mask_Upper'Result, Mask);
```

Reply:
That's bad. Please create at least a ticket for that issue.

Reply:
Fixed again, and hopefully right now :wink:

Reply:
It's perfect now. :slightly_smiling_face:   Thanks!